js/engine_unit_state.js

```diff
@@ -0,0 +1,448 @@
+ *
+ * Interfaces with the engine to retrieve information about current
+ * substance consumption, volumes, populations, time elapsed, and other
+ * metrics needed to convert between different unit types in the model.
+ */
+class ConverterStateGetter {
+  /**
+   * Create a new converter state getter.
+   *
+   * @param {Engine} engine - The engine instance to query for state information.
+   */
+  constructor(engine) {
+    const self = this;
+    self._engine = engine;
+  }
+
+  /**
+   * Get the consumption ratio per unit volume of substance.
+   *
+   * @returns {EngineNumber} The consumption per volume ratio in tCO2e/kg or tCO2e/mt.
+   * @throws {string} If consumption or volume units are not as expected.
+   */
+  getSubstanceConsumption() {
+    const self = this;
+    return self._engine.getEqualsGhgIntensity();
+  }
+
+  /**
+   * Get the energy consumption intensity per unit volume.
+   *
+   * @returns {EngineNumber} The energy intensity as a ratio (e.g., kwh/mt or kwh/kg).
+   * @throws {string} If the consumption or volume units are not as expected.
+   */
+  getEnergyIntensity() {
+    const self = this;
+    return self._engine.getEqualsEnergyIntensity();
+  }
+
+  /**
+   * Get the charge volume per unit for sales.
+   *
+   * @returns {EngineNumber} The charge volume in kg or mt per unit.
+   */
+  getAmortizedUnitVolume() {
+    const self = this;
+    return self._engine.getInitialCharge("sales");
+  }
+
+  /**
+   * Get the current equipment population.
+   *
+   * @returns {EngineNumber} The equipment count in units.
+   */
+  getPopulation() {
+    const self = this;
+    return self._engine.getStream("equipment");
+  }
+
+  /**
+   * Get number of yearsin the simulation since the last step.
+   *
+   * @returns {EngineNumber} The elapsed time in years since the last step.
+   */
+  getYearsElapsed() {
+    const self = this;
+    return new EngineNumber(1, "year");
+  }
+
+  /**
+   * Get the total ghg consumption for the current state.
+   *
+   * @returns {EngineNumber} The consumption value in tCO2e.
+   */
+  getGhgConsumption() {
+    const self = this;
+    return self._engine.getStream("consumption");
+  }
+
+  /**
+   * Get the total energy consumption for the current state.
+   *
+   * @returns {EngineNumber} The consumption value in kwh.
+   */
+  getEnergyConsumption() {
+    const self = this;
+    return self._engine.getStream("energy");
+  }
+
+  /**
+   * Get the total volume from sales for the current state.
+   *
+   * @returns {EngineNumber} The volume in kg or mt.
+   */
+  getVolume() {
+    const self = this;
+    const sales = self._engine.getStream("sales");
+    return sales;
+  }
+
+  /**
+   * Get the consumption ratio per unit of population.
+   *
+   * @returns {EngineNumber} The consumption per unit ratio in tCO2e/unit.
+   * @throws {string} If population or volume units are not as expected.
+   */
+  getAmortizedUnitConsumption() {
```

Comment:
Can you please update the JSDoc in engine_unit_state to clarify that AmortizedUnitVolume Refers to the amortized initial charge specifically?